### PR TITLE
Upgrade Flink

### PR DESCRIPTION
**Pull request by ranchodeluxe**
Addresses #149

NOTE: the [companion PR to upgrade the operator for the TF repo](https://github.com/pangeo-forge/pangeo-forge-cloud-federation/pull/6) will stay with the historyserver updates

### CODE REVIEW

File: .github/workflows/flink.yaml [hunk 1]
```diff
@@ -29,10 +29,9 @@ jobs:
       matrix:
         python-version: [ "3.9", "3.10", "3.11" ]
         recipes-version: [
-            "pangeo-forge-recipes==0.9.4",
+            "pangeo-forge-recipes==0.10.0",
```
Comment (cisaacstern) [inline .github/workflows/flink.yaml:32]: @ranchodeluxe generally I like deprecating 0.9.4 support (#144) but for my understanding, is there a reason this is part of this PR?

Comment (cisaacstern) [reply .github/workflows/flink.yaml:32]: And thanks for the lightening-fast turnaround on this!

Comment (ranchodeluxe) [reply .github/workflows/flink.yaml:32]: oh, haha, we can change it back so you can do the deprecation sorry (just thought I'd add it in)

**Review by cisaacstern — APPROVED** [review]
Thanks @ranchodeluxe, amazing!

**Pull request merged**